src/main/java/rx/singles/BlockingSingle.java

```diff
@@ -0,0 +1,56 @@
+package rx.singles;
+
+import rx.Single;
+import rx.SingleSubscriber;
+import rx.Subscription;
+import rx.internal.util.UtilityFunctions;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+public final class BlockingSingle<T> {
```

Comment:
This should start out as experimental unless the RxJava contributors want to fast-track this.